rust/storage/src/admissioncontrolleds3.rs

```diff
@@ -185,31 +227,40 @@ impl AdmissionControlledS3Storage {
         // request to S3.
         let future_to_await;
         {
-            let mut requests = self.outstanding_read_requests.lock();
+            let mut requests = self.outstanding_read_requests.lock().await;
             let maybe_inflight = requests.get(&key).cloned();
             future_to_await = match maybe_inflight {
                 Some(fut) => {
                     tracing::trace!("[AdmissionControlledS3] Found inflight request to s3 for key: {:?}. Deduping", key);
-                    fut
+                    fut.update_priority(options.priority).await;
```

Comment:
It seems like we could restructure this since update priority doesn't need to happen under the lock since we only ever increase priority ?